### ISSUE TITLE
Two scheduled notification created, only one is fired

### ISSUE THREAD

**Issue by usernamekk** [rafaelsetragni/awesome_notifications#226]
I created two notifications, online json and firebase but only one works at a time. Either the first works when the second is deleted or vice versa.

Secondly, I have multiple days through the year calendar scheduled notifications stored. In my code the notifications are created but only the last scheduled one is fired. Since all are created, can it be fired differently at a scheduled time?
Thanks as I wait for help.

```
Map<DateTime, List<EventList>> _events;
Map<DateTime, List<AddEventModel>> _events2;
  
void notify() async{
  //String localTimeZone = await AwesomeNotifications().getLocalTimeZoneIdentifier();
  var now = DateTime.now().subtract(Duration(hours: 1));

var v = _events.values.iterator;
var k = _events.keys.iterator;
while(v.moveNext() && k.moveNext()){
   if(k.current.isAfter(now) && v.current.single.program != null && v.current.single.program.length >0){

  var p =v.current.single.program.iterator;
     while(p.moveNext()){

    await AwesomeNotifications().createNotification(
        content: NotificationContent(
          id: p.current.programName.indexOf(p.current.programName),
          channelKey: 'Adlic',
          title: 'Awka Diocesan Event - Tomorrow (${DateFormat.yMMMEd().format(k.current)}, ${DateFormat.jms().format(k.current)})',
          body: p.current.programName,
          summary: 'Adlic',
          notificationLayout: NotificationLayout.BigPicture,
          bigPicture: 'asset://assets/images/logo.png',
          payload: {'uuid': 'uuid-adlic1'},
          autoCancel: false,
        ),
        schedule:
        //NotificationInterval(interval: 10, timeZone: localTimeZone, repeats: true));
        NotificationCalendar.fromDate(
            repeats: true,
            allowWhileIdle: true,
            date:
            DateUtils.utcToLocal(now.add(Duration(seconds: 20)))
            //DateUtils.utcToLocal(now.subtract(Duration(minutes: 59,seconds: 30)))
            // DateUtils.utcToLocal(k.current.subtract(Duration(hours: 7,minutes: 55,seconds: 0)))
           // DateUtils.utcToLocal(DateTime.now().subtract(Duration(minutes: 59, seconds: 30)))
          //DateTime(DateTime.now().year, DateTime.now().month, DateTime.now().day)
        )
    );
    print("First Time added more hours below");
   // print(k.current.subtract(Duration(hours: 16)));
    print(DateUtils.utcToLocal(now));
   // print(DateUtils.utcToLocal(k.current.subtract(Duration(hours: 7,minutes: 55,seconds: 0))));
       //  print(DateUtils.utcToLocal(now.subtract(Duration(minutes: 59,seconds: 30))));
  }}}  }


void notify2() async{
  var now = DateTime.now().subtract(Duration(hours: 1));
  //private events
  //if(_events2.){
  var pv = _events2.values.iterator;
  var pk = _events2.keys.iterator;

  while(pv.moveNext() && pk.moveNext()){
    // print(pv.current.single.title);
    // print(pk.current);
    if(pk.current.isAfter(now) && pv.current !=null && pv.current.length >0){



      await AwesomeNotifications().createNotification(
          content: NotificationContent(
            id: pv.current.indexOf(pv.current.single),
            channelKey: 'Adlic2',
            title: 'Your Events - Tomorrow (${DateFormat.yMMMEd().format(pv.current.single.datetime)}, ${DateFormat.jms().format(pv.current.single.datetime)})',
            body: '${pv.current.single.title}, \n${pv.current.single.desc.isNotEmpty ? pv.current.single.desc : ''}, @ \n${pv.current.single.loc.isNotEmpty ? pv.current.single.loc : ''}',
            summary: 'Adlic',
            notificationLayout: NotificationLayout.BigPicture,
            bigPicture: 'asset://assets/images/logo.png',
            payload: {'uuid': 'uuid-adlic2'},
            autoCancel: false,
            backgroundColor: Colors.teal[900].withBlue(100),
            color: Colors.teal[900].withBlue(100),
          ),
          schedule:
          //NotificationInterval(interval: 10, timeZone: localTimeZone, repeats: true));
          NotificationCalendar.fromDate(
              repeats: true,
              allowWhileIdle: true,
              date:
              DateUtils.utcToLocal(now.add(Duration(seconds: 90)))
            //  DateUtils.utcToLocal(pk.current.subtract(Duration(hours: 16)))
            // DateUtils.utcToLocal(DateTime.now().subtract(Duration(minutes: 59, seconds: 30)))
            //DateTime(DateTime.now().year, DateTime.now().month, DateTime.now().day)
          )
      );
      print("Time added more hours below");
      // print(pk.current.subtract(Duration(hours: 16)));
      print(DateUtils.utcToLocal(now));
    }}
}

```

**Comment by nicoaudy** [comment]
Same issue here

**Comment by rafaelsetragni** [comment]
If the notification has the same id, it will replace everything on the older one, including the schedule. You need to ensure to use different ids for each new schedule created.

For push notifications im developing right now a complementary plugin to handle it in place of `firebase_messaging`. This plugin called `awesome_notifications_fcm` and it will be available soon, with the release of version 0.0.7.

**Comment by rafaelsetragni** [comment]
Hi! Sorry for taking so long to respond to your issue on GitHub. I was focused on releasing the new 0.7.0 version of `awesome_notifications` and `awesome_notifications_fcm` and i didn't had time enough to do both. But now I can answer all your questions.

So now i'm asking you to recreate this topic using the new issue template. There's a lot of missing informations that i need to understand your problem, and all the instructions are already in the new issue template. Also, remember to check beforehand if your issue was posted by another user.

So I will automatically close all previous issues so far.
Sorry for the inconvenience and i will be waiting for your new issue request.

Thank you so much for your support!
